codegen/builtin_fs.go

```diff
@@ -91,13 +91,27 @@ func (i Image) Call(ctx context.Context, cln *client.Client, val Value, opts Opt
 	ref = reference.TagNameOnly(named).String()
 
 	var (
-		st       = llb.Image(ref, imageOpts...)
-		image    = &solver.ImageSpec{}
-		resolver = ImageResolver(ctx)
+		st         = llb.Image(ref, imageOpts...)
+		image      = &solver.ImageSpec{}
+		resolver   = ImageResolver(ctx)
+		resolveOpt = llb.ResolveImageConfigOpt{Platform: &platform}
 	)
 
+	// For some reason, llb.ResolveModeDefault defaults to
+	// llb.ResolveModeForcePull on BuildKit but it defaults to
+	// llb.ResolveModePreferLocal on docker engine.
```

Comment:
Any reason not to be explicit and unconditionally set `resolveOpt.ResolveMode = llb.ResolveModeForcePull.String()` regardless of what we expect the default to be?